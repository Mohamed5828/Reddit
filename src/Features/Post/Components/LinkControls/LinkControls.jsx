// Import styled components
import { LinkControlsContainer, StyledButton } from "./LinkControls.styled";

// Import icons
import { BsLink45Deg } from "react-icons/bs";
import { OverlayTrigger, Tooltip } from "react-bootstrap";

/**
 * The link control in the draft editor
 *
 * @param {Function} promptForLink - Function to show link form and prompt for link
 * @returns {React.Component} - The link control in the draft editor
 */
const LinkControls = ({ promptForLink }) => {
  return (
    <LinkControlsContainer>
      <OverlayTrigger
        placement="top"
        overlay={<Tooltip id="button-tooltip-2">Link</Tooltip>}
      >
        <StyledButton onMouseDown={promptForLink}>
          <BsLink45Deg size={25} />
        </StyledButton>
      </OverlayTrigger>
    </LinkControlsContainer>
  );
};

export default LinkControls;
