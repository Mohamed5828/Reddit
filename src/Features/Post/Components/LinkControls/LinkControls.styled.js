// Import styled
import styled from "styled-components";

export const LinkControlsContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const StyledButton = styled.span`
  cursor: pointer;
  margin-right: 5px;
  padding: 2px 0;
  display: inline-block;
  color: ${({ active, theme }) =>
    active ? theme.color.primary : theme.borderColor.primary};
  border-radius: 4px;
  outline: none;
  padding: 4px;
  &:hover {
    background-color: ${({ theme }) => theme.background.muted};
  }
`;
