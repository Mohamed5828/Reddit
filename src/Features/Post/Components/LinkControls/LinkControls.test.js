import { shallow } from "enzyme";
import React from "react";

// Import components
import LinkControls from "./LinkControls";

describe("Link controls of draft editor", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LinkControls />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
