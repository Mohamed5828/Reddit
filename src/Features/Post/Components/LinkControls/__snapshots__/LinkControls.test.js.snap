// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link controls of draft editor should render without crashing 1`] = `
<styled.div>
  <OverlayTrigger
    defaultShow={false}
    overlay={
      <Tooltip
        id="button-tooltip-2"
        placement="right"
      >
        Link
      </Tooltip>
    }
    placement="top"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <styled.span>
      <BsLink45Deg
        size={25}
      />
    </styled.span>
  </OverlayTrigger>
</styled.div>
`;
